README.md

```diff
@@ -83,172 +130,81 @@ Articles, on the other hand, support the following attributes:
-
-## Features
-
-### Caching
-
-The library provides a caching layer to avoid wasting data with very long feeds. All you need to do is provide an expiration date. For more details, give a look to the [Usage section](#usage)
-
-⚠️ For the time being caching is only supported if you are using Kotlin.
-
-### RSS Parsing from string
-
-The library provides a way of simply parsing raw data from a string.
-
-## Installation
-
-The library is uploaded on MavenCentral, so you can easily add the dependency:
-
-```Gradle
-dependencies {
-  implementation 'com.prof18.rssparser:rssparser:<latest-version>'
-}
-```
+  - Season
 
 ## Usage
 
-### Kotlin
+RssParser is available on the Android, iOS and JVM platforms and it's using [Coroutines](https://kotlinlang.org/docs/coroutines-overview.html) for all the asynchronous work.
 
-First of all, you have to create a `Parser` object with the Parser Builder. In the builder you can provided some custom and optional fields:
+### Creating a RssParser instance
 
-- A custom `OkHttpClient`; if not provided, the library will create one for you.
-- A custom Charset; if not provided, it will automatically inferred from the XML.
-- The Android Context, mandatory to make the caching layer work.
-- The cache expiration date in milliseconds, mandatory to make the caching layer work.
+To interact with the library, it's necessary first to create an instance of `RssParser`
 
-If you don't provide the Android Context and the cache expiration date, the library will continue working but without the caching feature.
+It's possible to create an instance of `RssParser` directly in the common code, without having to pass any platform-specific dependencies. 
 
-```Kotlin
-val parser = Parser.Builder()
-                .context(this)
-                .charset(Charset.forName("ISO-8859-7"))
-                .cacheExpirationMillis(24L * 60L * 60L * 1000L) // one day
-                .build()
+```kotlin
+val rssParser: RssParser = RssParser()
 ```
 
-The library uses the Kotlin Coroutines to retrieve, parse and cache the feed. So to start the process, you need to [launch](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/launch.html) the coroutine.
-
-```Kotlin
+#### Builder
 
-//url of RSS feed
-private val url = "https://www.androidauthority.com/feed"
-
-viewModelScope.launch {
-    try {
-        val channel = parser.getChannel(url)
-        // Do something with your data
-    } catch (e: Exception) {
-        e.printStackTrace()
-        // Handle the exception
-    }
-}
-```
+A `RssParser` instance can also be built with a platform-specific `RssParserBuilder`. In the builder, some custom and optional fields can be provided.
 
-For flushing the cache:
+**On Android and the JVM**, a custom `OkHttpClient` instance and a custom `Charset` can be provided. If an `OkHttpClient` instance is not provided, the library will create one. If no `Charset` is provided, it will automatically be inferred from the XML.
```

Comment:
```suggestion
**On Android and the JVM**, a custom `OkHttpClient` instance and a custom `Charset` can be provided. If an `OkHttpClient` instance is not provided, the library will create one. If no `Charset` is provided, it will automatically be inferred from the feed XML.
```